examples/custom-server/server.js

```diff
@@ -0,0 +1,33 @@
+const { createServer } = require('http')
+const { parse } = require('url')
+const next = require('next')
+const accepts = require('accepts')
+
+const app = next('.', { dev: true })
+const handle = app.getRequestHandler()
+
+app.prepare()
+.then(() => {
+  createServer((req, res) => {
+    const accept = accepts(req)
+    const type = accept.type(['json', 'html'])
+    if (type === 'json') {
```

Comment:
`<Link href="/foo" as="/bar">` is for caching. A JSON response for a component is always identical, so we want to allow only `<Link href="/blog" as "/blog/123">` instead of `<Link href="/blog/123">`